.github/actions/perform-deployment/index.js

```diff
@@ -0,0 +1,123 @@
+// Copyright 2014-2021 Scalyr Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const core = require('@actions/core');
+const cache = require('@actions/cache');
+const fs = require('fs');
+const path = require('path')
+const child_process = require('child_process')
+const buffer = require('buffer')
+const readline = require('readline')
+
+
+async function run() {
+  try {
+    const deploymentName = core.getInput("deployment-name")
+    const cacheVersionSuffix = core.getInput("cache-version-suffix")
+    const cacheDir = "deployment_caches"
+
+    // Get json list with names of all deployments which are needed for this deployment.
+    const deployment_helper_script_path = path.join("agent_build" ,"scripts", "run_deployment.py")
+    // Run special github-related helper command which returns names for all deployments, which are used in the current
+    // deployment.
+    const code = child_process.execFileSync(
+        "python3",
+        [deployment_helper_script_path, "deployment", deploymentName, "get-deployment-all-cache-names"]
+    );
+
+    // Read and decode names from json.
+    const json_encoded_deployment_names = buffer.Buffer.from(code, 'utf8').toString()
+    const deployer_cache_names = JSON.parse(json_encoded_deployment_names)
+
+    const cache_hits = {}
+
+    // Run through deployment names and look if the is any existing cache for them.
+    for (let name of deployer_cache_names) {
```

Comment:
Same here, may be easier to follow / test it if this block is in a separate utility function (e.g. ``getCacheInfoFromFile`` or similar).